### PR TITLE
LPD-18243 Data Set View Action icon button doesn't show completely if an icon is not selected

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPD-18243 Data Set View Action icon button doesn't show completely if an icon is not selected](https://issues.liferay.com/browse/LPD-18243)
- This is a followup on https://github.com/liferay-frontend/liferay-portal/pull/4038

### What is the goal of this PR?

We are rendering action column at the size of action button, in the case it is getting rendered.

I tried the solution I suggested in  https://github.com/liferay-frontend/liferay-portal/pull/4038#discussion_r1512896771, but I didn't really like it. It ended up needing to hardcode-compensate for padding widths, border widths and boldness of font. All of that means style modifications from customers may easily break this. Instead, I came up with an alternative solution, see inline comment.

There is a lot on noise in this PR, as I migrated to TS bunch of stuff. There were relevant in solution I gave up on, but I left in TS migration, because why not. I'll highlight the most relevant code, that fixes the bug.

cc @ugeortiz

### What does it look like?

screen2
screen3
screen1

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+								monospaced={false}
+								style={{
+									'visibility': 'hidden',
+									'white-space': 'nowrap',
+								}}
+							>
+								{itemsActions[0].label}
+							</LinkOrButton>
+						)}
+				</DndTable.Cell>
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.tsx:147]: This is the main change in this PR. The solution works like this:
1. On first render:
   - Widths are not set to `styles` yet. `isFixed` is the flag that is set once widths are set. 
   - We temporarily render a button, containing the label, in the header. 
   - The button is not visible, but still keeps its layout dimensions, thanks to how `visibility: 'hidden'` works.
   - We set the `defaultWidth` to 0, so the width is determined by the content of the cell.
2. On second render:
   - Widths are set, `isFixed` is `true`
   - We no longer show button in DOM

Notes:
- Although it's still possible that customers' styles break this, with very specific CSS selectors, it's not very likely.  This is probably good enough.
- There is a case that this solution does not solve, where items have different sets of actions, and some of them are with one actions. That case, I'm fairly sure, is not possible to solve without glitching table on data arrival. There is a workaround for it though, the `uniformActionsDisplay` prop, that renders ellipsis on everything.

**Comment by markocikos** [comment]
ci:test:relevant

**Comment by herrardo** [comment]
From concept perspective sounds good @markocikos gj!

**Comment by juanjofgliferay** [comment]
Awesome job! :clap:

**Comment by markocikos** [comment]
ci:forward

**Pull request closed without merging**